flh_opt/api_opt.py

```diff
@@ -0,0 +1,226 @@
+            max_hours=24,  # TODO: move this parameter out of the code.
+            cyclic_state_of_charge=True,
+            marginal_cost=input_data[name]["OPEX_O"],
+            p_nom_extendable=True,
+        )
+
+    add_storage(n, input_data, "EL_STR", "ELEC")
+    add_storage(n, input_data, "H2_STR", "H2")
+
+    # add RE profiles:
+    res_profiles = pd.DataFrame()
+    for g in input_data["RES"]:
+        process_code = g["PROCESS_CODE"]
+        if len(input_data["RES"]) > 1:
+            re_location = "RES_HYBR"
+        else:
+            re_location = process_code
+        res_profiles[process_code] = get_profiles(
+            source_region_code=input_data["SOURCE_REGION_CODE"],
+            process_code=process_code,
+            re_location=re_location,
+            selection=range(0, 48),  # TODO: make this a function parameter?
+        )["specific_generation"]
+
+    # define snapshots:
+    n.snapshots = res_profiles.index
+
+    # define snapshot weightings:
+    n.snapshot_weightings["generators"] = 8760.0 / len(n.snapshots)
+    n.snapshot_weightings["objective"] = 8760.0 / len(n.snapshots)
+    n.snapshot_weightings["stores"] = 1
+
+    # import profiles to network:
+    n.import_series_from_dataframe(res_profiles, "Generator", "p_max_pu")
+
+    # solve optimization problem:
+    n.optimize(solver_name="highs")
+
+    # calculate results:
+
+    def get_flh(n: Network, g: str, component_type: str) -> float:
+        if component_type == "Generator":
+            flh = n.generators_t["p"][g].mean() / n.generators.at[g, "p_nom_opt"]
+        if component_type == "Link":
+            flh = n.links_t["p0"][g].mean() / n.links.at[g, "p_nom_opt"]
+        return flh
+
+    result_data = {}
+    result_data["RES"] = []
+
+    # Calculate total RES capacity:
+    list_res = [item["PROCESS_CODE"] for item in input_data["RES"]]
+    cap_total = n.generators.loc[list_res, "p_nom_opt"].sum()
+
+    # Add results for each RES type:
+    for g in input_data["RES"]:
+        d = {}
+        d["PROCESS_CODE"] = g["PROCESS_CODE"]
+        d["FLH"] = get_flh(n, g["PROCESS_CODE"], "Generator")
+        d["SHARE_FACTOR"] = n.generators.at[g["PROCESS_CODE"], "p_nom_opt"] / cap_total
+        result_data["RES"].append(d)
+
+    # Calculate FLH for electrolyzer:
+    result_data["ELY"] = {}
+    result_data["ELY"]["FLH"] = get_flh(n, "ELY", "Link")
+
+    # calculate capacity factor for storage units:
+    # TODO: we use storage capacity per output, is this correct?
+    # TODO: or rather use p_nom per output?
+    result_data["EL_STR"] = {}
+    result_data["EL_STR"]["CAP_F"] = (
+        n.storage_units.at["EL_STR", "p_nom_opt"]
+        * n.storage_units.at["EL_STR", "max_hours"]
+    )
+    result_data["H2_STR"] = {}
+    result_data["H2_STR"]["CAP_F"] = (
+        n.storage_units.at["H2_STR", "p_nom_opt"]
+        * n.storage_units.at["H2_STR", "max_hours"]
+    )
+    return result_data, n
```

Comment:
Why do we need to return the network?

Reply:
In production, we may not need to return the network, but I did now because it makes inspection and testing easier